Small fixes in spectrum_file

cosama opened the pull request:
Fixes #159

- [x] Changed `UserWaring` to `SpectrumFileParsingWarning` in spectrum_file.py
- [x] removed unused member `energy_bin_widths`
- [x] changed `energy_bin_edges` to `bin_edges_kev`
- [x] Adjusted `spectrum.py`
- [x] Fix parser doc strings

jccurtis commented on becquerel/parsers/spectrum_file.py:
```diff
@@ -134,9 +134,3 @@ def energy_to_channel(self, energy):
-        """Calculate the width of the bin in keV at the channel(s)."""
-        en0 = self.channel_to_energy(channel - 0.5)
-        en1 = self.channel_to_energy(channel + 0.5)
-        return en1 - en0
```
I don't think we should remove this. This may be used by some other analysis code someone is using and would break that usage. We should think about refactoring in `1.0.0`

jccurtis commented on becquerel/parsers/README.markdown:
```diff
@@ -17,7 +17,7 @@ Other data available include:
 * `spec.livetime`
 * `spec.realtime`
-* `spec.energy_bin_widths`
+* `spec.bin_edges_kev`
```
```suggestion
* `spec.bin_edges_kev`
* `spec.energy_bin_widths` (deprecated)
```

jccurtis commented on becquerel/parsers/cnf_file.py:
```diff
@@ -93,7 +93,7 @@ class CnfFile(SpectrumFile):
         spec.channels
         spec.energies
-        spec.energy_bin_widths
+        spec.bin_edges_kev
```
```suggestion
        spec.bin_edges_kev
        spec.energy_bin_widths (deprecated)
```

jccurtis commented on becquerel/parsers/spc_file.py:
```diff
@@ -24,7 +24,7 @@ class SpcFile(SpectrumFile):
         spec.channels
         spec.energies
-        spec.energy_bin_widths
+        spec.bin_edges_kev
```
```suggestion
        spec.bin_edges_kev
        spec.energy_bin_widths (deprecated)
```

jccurtis commented on becquerel/parsers/spe_file.py:
```diff
@@ -32,7 +32,7 @@ class SpeFile(SpectrumFile):
         spec.channels
         spec.energies
-        spec.energy_bin_widths
+        spec.bin_edges_kev
```
```suggestion
        spec.bin_edges_kev
        spec.energy_bin_widths (deprecated)
```

jccurtis commented on becquerel/parsers/spectrum_file.py:
```diff
@@ -29,7 +29,7 @@ class SpectrumFile(object):
         spec.channels
         spec.energies
-        spec.energy_bin_widths
+        spec.bin_edges_kev
```
```suggestion
        spec.bin_edges_kev
        spec.energy_bin_widths (deprecated)
```

jccurtis commented on becquerel/parsers/spectrum_file.py:
```diff
@@ -56,8 +56,7 @@ def __init__(self, filename):
         self.cal_coeff = []
         # arrays to be calculated using calibration
         self.energies = np.array([], dtype=np.float)
-        self.energy_bin_widths = np.array([], dtype=np.float)
```
Don't remove. You can change this to `bin_widths_keV`, create an alias property for `energy_bin_widths` and have the `bin_widths_keV` property raise a `DeprecationWarning`

cosama commented:
@jccurtis Good point, I agree we should not break stuff at this point. I added getters for the two members that I removed, one should be marked as depreciated now. I don't think we should add a `bin_widths_kev` member before we all agree that we like to have that also in the future and that we might also want to add it to the spectrum class. I can life with `energy_bin_widths` being slightly off the naming convention.

cosama commented:
Also I just realized `DeprecationWarnings` are not shown unless run `python` is run with the `-W` flag. It is meant to inform developers that run their code with that flag on to make sure stuff is not changing. As `becquerel` is used mostly by students that should be informed about upcoming changes it might make sense to turn the flag on in the code. Probably would trigger a whole bunch of other message though. Any thoughts?

jccurtis commented:
@cosama lets figure out what changes need to be made for this. We will also need to update this branch ...

markbandstra commented on becquerel/parsers/README.markdown:
```diff
@@ -17,8 +17,9 @@ Other data available include:
+* `spec.bin_edges_kev`
 * `spec.energy_bin_widths`
-
+* `spec.energy_bin_edges` (depricated)
```
check spelling

jccurtis commented:
thanks @cosama!
